### DISCUSSION TITLE
Close a toast by id

### DISCUSSION THREAD

**Discussion by Et3rnal** [primefaces/community#2872] [PrimeNG]
I have a single p-toast on my `app.component.html` and add message using the messages service only across the app without a key

I noticed there is no way to close/clear a single message, the only way is to introduce another p-toast just to be able to close that message, since messages has id it would be nice to add a close by id method that doesn't we can use without a key

**Comment by thepuzzlemaster** [comment]
I also have a use-case which runs into this problem.

I have 2 different toasts that can be rendered in the same component. One uses a regular toast, and one uses a custom templated toast. However, I would like the ability to programmatically close either toast.

Each toast could spawn more than 1 instance at the same time, so I don't believe there is a way to specify which toast I'd like to close.

e.g. my template (simplified) looks something like this.
```
<p-toast key='toastA' />
<p-toast key='toastB' />
```

So when I use `messageService.add()`, I need to use the `key` to reference which toast should spawn the message, and they cannot be unique across instances. There is no way for me to close the toast programmatically.

**Comment by thepuzzlemaster** [comment]
I'm not stoked about this solution, because it relies on private implementation details of the Toast component, but it works.

my-component.ts
```
@Component({
  selector: 'my-component',
  template: "<p-toast #toaster key='toaster' />",
  imports: [ToastModule],
})
export class MyComponent {
  @ViewChild('toaster') toast: Toast | null = null;

  constructor (private messageService: MessageService) {}

  protected createToast(toastId: string) {
    this.messageService.add({
      key: 'toaster',
      id: toastId,
      // ...other properties
    })
  }  

  protected closeToast(toastId: string) {
      const toastIndex = this.toaster?.messages?.findIndex(message) => message.id === toastId ?? -1;
      const message = this.toaster?.messages?.find(message) => message.id === toastId);

      if (message && toastIndex !== -1) {
        this.toaster?.onMessageClose({ index: toastIndex, message });
      }
    }
}
```
